ubcpi/answer_pool.py

```diff
@@ -0,0 +1,315 @@
+import random
+import persistence as sas_api
+
+POOL_SIZE = 50
+
+
+class UnknownChooseAnswerAlgorithm(Exception):
+    pass
+
+
+def offer_answer(pool, answer, rationale, student_id, algo):
+    """
+    submit a student answer to the answer pool
+
+    The answer maybe selected to stay in the pool depending on the selection algorithm
+
+    Args:
+        pool (dict): answer pool
+            Answer pool format:
+            {
+                option1_index: {
+                    'student_id': { can store algorithm specific info here },
+                    ...
+                }
+                option2_index: ...
+            }
```

Comment:
Yes, the structure is correct. The number of responses stored can be depending on the algorithm. The two current implemented ones are storing 50 for each option as I indicated in another comment when calculating the side. The size is multiplied by the number of options. So if you have 10 options, it will be 500. The limit of 50 is for each option. If you feel it is too big, we can easily reduce the number from 50 to a lower value. It is defined as a constant.